### ISSUE TITLE
Validation and test set seems to be the same ?

### ISSUE THREAD

**Issue by koushiksrivats** [wangzhuo2019/SSAN#14]
Dear Authors
Thanks for the great work.

I was going through your code to understand how you have used each of the 4 datasets for cross dataset training and testing.

**Observations:**
1. As per the code in the [data_merge](https://github.com/wangzhuo2019/SSAN/blob/f4c2cbcd351a57766fc2ee7b83bde620c5cb00fc/datasets/data_merge.py#L60) file, you are loading the target dataset as your test set (which seems fine).

2. However, you seem to be using the target dataset as your validation set to compute the HTER and AUC scores after every epoch, and choose the epoch with the best score on the target dataset. [training file for reference](https://github.com/wangzhuo2019/SSAN/blob/f4c2cbcd351a57766fc2ee7b83bde620c5cb00fc/solver.py#L125).  **Example**: _Lets choose OCIM protocol. For every epoch, you seem to be training on OCI and testing on M and choosing the epoch with the best score on M._

**Questions:**
- Is this approach valid ? Shouldn't there be a validation set combining  O-C-I, that is used to evaluate during training and finally the chosen model is tested on M to compute the HTER and AUC scores ?

Kindly request you to clarify the same.

Thanks

**Comment by yeldartoktasynov** [comment]
I have same confusion, so the same question.
